docs/DeveloperGuide.md

```diff
@@ -1,34 +1,413 @@
+        An integer that represents the index of the `Note` object to be deleted within the `NoteList` is passed into 
+        this method.
+
+Step 2: The `NoteList.delete()` method makes use of `ArrayList` API, specifically the `ArrayList.remove()` method, to 
+        delete the `Note` object in `NoteList`.
+
+**Listing out all notes in a topic**
+
+This task is performed by the `Ui.printNoteList()` method.
+
+Step 1: The `parseCommand()` method instantiates a `NoteCommand` object which then instantiates an Ui object
+and calls the `Ui.listInteraction` method. 
+
+Step 2: The `Ui.listInteraction` method calls the `Ui.printNoteList()` method. The topic's `NoteList` is passed into 
+this method. `Ui.printNoteList()` prints out the descriptions and texts of all the `Note` objects in the 
+topic's `NoteList`.
+
+#### 2.2.4. Design of Option and OptionList 
+
+The `Option` and `OptionList` classes implements the `Displayable` and `DisplayableList` interfaces respectively. 
+The `Option` object stores one option of a question while the `OptionList` object stores all 4 options of the same 
+question. The class diagram below illustrates the structure of both classes. 
+
+![Option_and_OptionList_Class](./images/Option.png)
+
+### 2.3. Logic Component
+
+The main application logic, such as provisioning quizes, is handled by the Logic component. This component also acts as the middleman between the backend and frontend by processing data before passing it to the user interface and parsing user input from the user interface.
+
+#### 2.3.1. Design of Parser
+
+![Parser Diagram](./images/ParserDiagram.png)
+
+1. After constructing a new MenuParser() in the `Eduke8` class, the parseCommand() method is used to parse the 
+   user command.
+2. This results in a `Command` object, which is executed by `Command` class itself, using the execute() method.
+3. The `Ui` object in the `Command` object is used to display the requested information, or to display the required task 
+to be completed as per the user input.
+
+#### 2.3.2. Implementation of MenuParser
+
+Below is the sequence diagram for how the Parser component of `Eduke8` works with commands to show output to the user.
+
+![Parser Sample Sequence](./images/ParserSampleSequence.png)
+
+The command parsing feature is our program’s way of reading the user’s input into the command line. It makes use of a 
+single method `parseCommand` that identifies what command the user is calling for and then calls the command. There are 
+two parsers in our program that implements a single `Parser` interface. One parser is for choosing menu options and is 
+named `MenuParser`. The other parser is used during quizzes, in order to answer questions or request for hints, and is 
+called `QuizParser`. Given below is an example usage scenario of how the command parsing feature works at each step, 
+when the user types in input to get help in order to see what commands are available to the user.
```

Comment:
A well written explanation, but this text appears as a rather lengthy paragraph when viewed in the IO. Perhaps some formatting such as bulleting could be used to make it easier for the reader to understand the key points of the paragraph.